File: content-solutions/dfs/team-size.py
def solution(boss):
    n = len(boss)
    children = [[] for _ in range(n)]
    root = 0
    for i, b in enumerate(boss):
        if b == -1:
            root = i
        else:
            children[b].append(i)
    sizes = [0] * n

    def dfs(v):
        total = 1
        for w in children[v]:
            total += dfs(w)
        sizes[v] = total
        return total

    dfs(root)
    return sizes

File: content-solutions/dfs/team-size.js
function solution(boss) {
  const n = boss.length;
  const children = Array.from({ length: n }, () => []);
  let root = 0;
  boss.forEach((b, i) => {
    if (b === -1) root = i;
    else children[b].push(i);
  });
  const sizes = new Array(n).fill(0);
  function dfs(v) {
    let total = 1;
    for (const w of children[v]) total += dfs(w);
    sizes[v] = total;
    return total;
  }
  dfs(root);
  return sizes;
}
